### ISSUE TITLE
macOS AMD executable 1.24 is not working right.

### ISSUE THREAD

**Issue by tempelmann** [icloud-photos-downloader/icloud_photos_downloader#977]
I am running build 1.24 on macOS 10.13.6 (High Sierra). The executable runs fine, I can even login with using the system Keychain for storing the password:

    $ /Users/tempi/Downloads/icloudpd --version
    version:1.24.0, commit sha:0e608f6, commit timestamp:Fri Oct 25 18:18:53 2024 CEST

But the help pages are all misleading, making it difficult to understand how to use this tool to simply download all files to a local directory.

For instance, the main readme says:

>  (use --help option to get full list)

But "--help" only gives an incomplete list and not a full list of all cmds:

```
$ /Users/tempi/Downloads/icloudpd --help
Usage: icloudpd <options>

Options:
  --username TEXT        Apple ID to Use
  --password TEXT        Apple ID Password to Use; if unspecified, password
                         will be fetched from the system keyring.
  -n, --non-interactive  Disable interactive prompts.
  --delete-from-keyring  Delete stored password in system keyring for this
                         username.
  --domain [com|cn]      Root Domain for requests to iCloud (com or cn)
                         [default: com]
  --version              Show the version, commit hash and timestamp
  -h, --help             Show this message and exit.
```

This does NOT list all commands as suggested.

When I run the tool without cmds:

```
usage: icloudpd [-h] [--username USERNAME] [--password PASSWORD] [-n] [--delete-from-keyring] [--list] [--llist]
                [--locate] [--device DEVICE_ID] [--sound] [--message MESSAGE] [--silentmessage SILENTMESSAGE]
                [--lostmode] [--lostphone LOST_PHONE] [--lostpassword LOST_PASSWORD] [--lostmessage LOST_MESSAGE]
                [--outputfile] [--domain DOMAIN]
```

This is wrong in several ways:

1. It does not even mention that one should provide a path (I figured that out only from looking at examples elsewhere)
2. cmds like "list" and "llist" are not even available, even if I also provide the login info:

```
$ /Users/tempi/Downloads/icloudpd --list  --username xx@me.com
Usage: icloudpd <options>
Try 'icloudpd -h' for help.

Error: No such option: --list
````

I then also checked the docs, e.g. https://icloud-photos-downloader.github.io/icloud_photos_downloader/reference.html

That lists commands such as "list-albums", but again those don't seem to work:

```
$ /Users/tempi/Downloads/icloudpd --username xx@me.com --list-albums
Usage: icloudpd <options>
Try 'icloudpd -h' for help.

Error: No such option: --list-albums
```

Even the example given in the Readme isn't working:

```
$ /Users/tempi/Downloads/icloudpd --directory /tmp  --username xx@me.com --watch-with-interval 3600
Usage: icloudpd <options>
Try 'icloudpd -h' for help.

Error: No such option: --directory
```

Basically, I cannot get this tool to work at all because nothing seems to work.

Or course, I have no trouble getting my photos synched with Apple's own icloud services, as I can see them in the Photos.app.

What's going on here?

**Comment by tempelmann** [comment]
Oh, wow: The above "issue" only is with the direct download from the releases page. I downloaded https://github.com/icloud-photos-downloader/icloud_photos_downloader/releases/download/v1.24.0/icloud-1.24.0-macos-amd64

However, if I install "icloudpd" with the macOS "brew" command (homebrew), then the usage and help is working!

The version from the brew install says:

    $ icloudpd --version
    version:1.24.0, commit sha:0e608f6, commit timestamp:Fri Oct 25 18:18:53 2024 CEST

Oddly, they both appear to be the same version. What's going on here?

So I compare the two executables...

The one I downloaded from the releases is a pure executable. The one installed with brew is a python script, and in its dir (/usr/local/Cellar/icloudpd/1.24.0/libexec/bin) I find other tool such as keyring, flask and tqdm.

So, even though both versions  appear to be the same, according to their "--version" info, they behave differently. I wonder if this should be considered a bug in the deployment procedure.

**Comment by AndreyNikiforov** [comment]
Based on the set of parameters, you are downloading `icloud`, not `icloudpd` (but renaming it to icloudpd). Use full file names as downloaded for easier troubleshooting.

Downloaded from github release to M3 Air with Sonoma 14.6.1 and got help correctly:
```
removed@removed ~ % ~/Downloads/icloudpd-1.24.0-macos-amd64 --help
Usage: icloudpd-1.24.0-macos-amd64 <options>

  Download all iCloud photos to a local directory

Options:
  -d, --directory <directory>     Local directory that should be used for
                                  download
  -u, --username <username>       Your iCloud username or email address
  -p, --password <password>       Your iCloud password (default: use PyiCloud
                                  keyring or prompt for password)
  --auth-only                     Create/Update cookie and session tokens
                                  only.
  --cookie-directory </cookie/directory>
                                  Directory to store cookies for
                                  authentication (default: ~/.pyicloud)
  --size [original|medium|thumb|adjusted|alternative]
                                  Image size to download. `medium` and `thumb`
                                  will always be added as suffixes to
                                  filenames, `adjusted` and `alternative` only
                                  if conflicting, `original` - never. If
...
```

**Comment by tempelmann** [comment]
Oops!
